config/jobs/ocp-automation/openshift-install-power/openshift-install-power-images.yaml

```diff
@@ -0,0 +1,93 @@
+            volumeMounts:
+              - name: kubeconfig
+                mountPath: /etc/kubeconfig
+            command:
+              - /bin/bash
+            args:
+              - -c
+              - |
+                set -o errexit
+                set -o nounset
+                set -o pipefail
+                pushd $GOPATH/src/github.com/vmware-tanzu/buildkit-cli-for-kubectl
+                make build install
+                popd
+                ca=$(base64 -w 0 /run/secrets/kubernetes.io/serviceaccount/ca.crt)
+                token=$(cat /run/secrets/kubernetes.io/serviceaccount/token)
+                namespace=$(cat /run/secrets/kubernetes.io/serviceaccount/namespace)
+                cat >kubeconfig <<EOL
+                apiVersion: v1
+                kind: Config
+                clusters:
+                - name: default-cluster
+                  cluster:
+                    certificate-authority-data: ${ca}
+                    server: https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT_HTTPS}
+                contexts:
+                - name: default-context
+                  context:
+                    cluster: default-cluster
+                    namespace: ${namespace}
+                    user: default-user
+                current-context: default-context
+                users:
+                - name: default-user
+                  user:
+                    token: ${token}
+                EOL
+                export KUBECONFIG=$(pwd)/kubeconfig
+                for version in 4.5 4.6 4.7
+                do
+                    kubectl build --push --registry-secret quay-powercloud-regcred --namespace image-builder --build-arg RELEASE_VER=$version -t quay.io/powercloud/openshift-install-powervs:ocp$version-amd -t quay.io/powercloud/openshift-install-powervs:$PULL_BASE_REF-ocp$version-amd -f images/Dockerfile ./
+                    kubectl build --push --registry-secret quay-powercloud-regcred --kubeconfig /etc/kubeconfig/config --namespace image-builder --build-arg RELEASE_VER=$version -t quay.io/powercloud/openshift-install-powervs:ocp$version-ppc64le -t quay.io/powercloud/openshift-install-powervs:$PULL_BASE_REF-ocp$version-ppc64le -f images/Dockerfile ./
+                done
+                curl -L https://github.com/estesp/manifest-tool/releases/download/v1.0.3/manifest-tool-linux-`go env GOARCH` -o /usr/local/bin/manifest-tool
+                chmod +x /usr/local/bin/manifest-tool
+                cat > config.json << EOL
+                $DOCKER_CONFIG
+                EOL
+                for version in 4.5 4.6 4.7
+                do
+                cat > registry.yaml <<EOL
+                image: quay.io/powercloud/openshift-install-powervs:ocp$version
+                tags: ['$PULL_BASE_REF-ocp$version', 'latest']
+                manifests:
+                  -
+                    image: quay.io/powercloud/openshift-install-powervs:ocp$version-ppc64le
```

Comment:
@yussufsh is this image already created in the org? I don't see one.

Reply:
I am about to run this now. It might create the images if everything is fine.
I don't think images are already created. I too don't see any.

Reply:
No, we need to push it via this job.

Reply:
We need to create an empty image and map the robot ID for the image push, which I have done now. @Rajalakshmi-Girish please run end to end and check if it is really pushing the image or not.

Reply:
> 
> 
> > > /cc @Rajalakshmi-Girish need your help in review and testing this change.
> > > /cc @mkumatag
> > 
> > 
> > @mkumatag Please let me know if I can `test-pj` with a test-pod on iks prow-cluster? I do not have a x86 cluster.
> > @yussufsh I see the versions  4.5 4.6 4.7 hardcoded. Can we not have the images for older versions built manually? I thought this post-submit job was meant for the building images when there is a new version.
> > Please correct me if I understand it wrong!
> 
> Go ahead...

@mkumatag while I run the job, I am unable to view the logs of `test-pod` due to the issue we have with iks `prow-cluster` for which we have already raised a support case [CS2300752](https://cloud.ibm.com/unifiedsupport/cases?number=CS2300752).
Other than uploading the logs to s3 storage (I assume it is fine to do), Can you please suggest if there is any other way of viewing it.

Reply:
> @mkumatag while I run the job, I am unable to view the logs of `test-pod` due to the issue we have with iks `prow-cluster` for which we have already raised a support case [CS2300752](https://cloud.ibm.com/unifiedsupport/cases?number=CS2300752).
> Other than uploading the logs to s3 storage (I assume it is fine to do), Can you please suggest if there is any other way of viewing it.

not sure how to tackle this.!

Reply:
@mkumatag instead of having the logs on local mode, I pushed them to s3 storage.
But the job has been in pending status for more than an hour now.
https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/openshift-install-power-build-and-push-on-release/1397136614659985408
I only see `clone-log.txt` as the job isn't completed yet. Unable to fetch build-log :(
The `describe pod` doesn't show any error!
Should we think of having a new cluster on some x86 VM?

Reply:
The job timed out after 2hrs.
[build-log.txt](https://github.com/ppc64le-cloud/test-infra/files/6539593/build-log.txt)
```
/home/prow/go/src/github.com/vmware-tanzu/buildkit-cli-for-kubectl /home/prow/go/src/github.com/ocp-power-automation/openshift-install-power
GOOS=linux go build -ldflags "-X github.com/vmware-tanzu/buildkit-cli-for-kubectl/version.Version=v0.1.2-0-g25b70ef" -mod=vendor -o bin/linux/kubectl-buildkit ./cmd/kubectl-buildkit
GOOS=linux go build -ldflags "-X github.com/vmware-tanzu/buildkit-cli-for-kubectl/version.Version=v0.1.2-0-g25b70ef" -mod=vendor -o bin/linux/kubectl-build  ./cmd/kubectl-build
cp ./bin/linux/kubectl-buildkit ./bin/linux/kubectl-build /usr/local/bin
/home/prow/go/src/github.com/ocp-power-automation/openshift-install-power
time="2021-05-25T10:27:05Z" level=error msg="error dialing backend: read tcp 192.168.255.6:46014->10.240.129.7:10250: read: connection reset by peer"
{"component":"unset","file":"/go/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 2h0m0s timeout","severity":"error","time":"2021-05-25T12:25:36Z"}
{"component":"unset","file":"/go/test-infra/prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15s grace period","severity":"error","time":"2021-05-25T12:25:51Z"}
```

Reply:
Looking into it. Seems there is some problem talking with buildkit pod on the 10.240.129.7 node of iks cluster.

Reply:
I just tried to check if things are fine with ppc64le architecture.
The image got built successfully and pushed. Here is the job:
https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/openshift-install-power-build-and-push-on-release/1397422136011788288